[HWKINVENT-22] Make dist use inventory with Titan.

metlos opened the pull request:
For this to work in dist, it requires the common embedded cassandra EAR to be deployed in Nest.

Inventory will use the Cassandra instance for data storage and will start an embedded, publicly unavailable ElasticSearch instance for indexing.

Right now, the Titan configuration is embedded inside the integrated war file, but there is a TODO to load it from somewhere more meaningful ;).

pilhuhn commented:
On 10 Jun 2015, at 21:31, Lukas Krejci wrote:

> Inventory will use the Cassandra instance for data storage and will 
> start an embedded, publicly unavailable ElasticSearch instance for 
> indexing.

That sounds pretty fat and expensive?

metlos commented:
It can be later easily changed to use a shared (in VM or remote) ES instance with BTM, but I didn't want to spend time setting that up until we're at a point when it's actually needed. Thing is Titan needs an external index to be reasonably performant and even more importantly to be able to use operators like > or < on property values.

jkremser commented:
Is it ready to merge? Has "embedded C\* EAR" already landed in the Nest ;) I was looking there, and I can see the `hawkular-metrics-embedded-cassandra-ear.ear` so this should be ok. I am trying to run it..

jkremser commented:
when running `hawkular/hawkular@inventory-next` with locally built `hawkular/hawkular-inventory@pr85` (titan), it still stores the graph into `data/hawkular-inventory/tinkergraph.dat`. Not sure what I forgot.. the version of metrics is `0.3.2`, but the same thing happens with `0.3.4`, `0.3.5` is not in the mvn repo, and I didn't try with `0.3.5-SNAPSHOT`

pilhuhn commented:
Note that embedded C\* has been moved to hawkular-commons.

pilhuhn commented on hawkular-integrated-inventory-rest/src/main/resources/hawkular-inventory.properties:
```diff
@@ -0,0 +1,25 @@
+# limitations under the License.
+#
+
+storage.backend=cassandra
```
Please a) synchronize that with other sub-projects.
See e.g. http://www.hawkular.org/docs/components/metrics/configuration.html
and
allow to set hostname + port via envronment variables (as indicated in above url).
Also HAWKULAR_BACKEND env variable is used in embedded cassandra to decide if this runs or not. 
Not sure if the above is related to it or rather switches between sql/C\* backends.

pilhuhn commented on hawkular-integrated-inventory-rest/src/main/java/org/hawkular/integrated/inventory/InventoryConfigurationProducer.java:
```diff
@@ -0,0 +1,48 @@
+        props.put("jboss.server.data.dir", System.getProperty("jboss.server.data.dir"));
+
+        return new TokenReplacingReader(new InputStreamReader(getClass().getClassLoader()
+                .getResourceAsStream("hawkular-inventory.properties"), "UTF-8"), props);
```
As written below - please allow some overrides via ENV variables.

metlos committed:
[HWKINVENT-22] Make dist use inventory with Titan.

This also adds the ability to retry transactions which is useful
in case of concurrent access and failures stemming from concurrent
updates.

Also, make Titan use the same version of HTTPClient as our Pinger
uses to workaround the fact that Pinger now inherits from the
inventory's classpath.

metlos commented:
I removed (most of) the configuration aspect from the PR (and will open a new one for that). I also squashed the commits into a single "atomic" one, because it is quite well isolated piece of change.

ppalaga commented:
lgtm
